app/src/main/java/com/example/myapplication/SecondFragment2.java

```diff
@@ -311,6 +312,7 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
             public void onClick(View v) {
                 if (MainActivity.notesThrown > 0) {
                     MainActivity.notesThrown--;
```

Comment:
Similar comment as previous file - no need to store data in MainActivity anymore:

    MatchData matchData = UserModel.getMatchData();
    matchData.setHumanPlayerNotesThrown(matchData.getHumanPlayerNotesThrown() - 1);

Reply:
Same for subsequent changes